request/src/main/java/com/vimeo/networking2/Configuration.kt

```diff
@@ -0,0 +1,272 @@
+ * index.
+ * @param accountStore The store used to remember authenticated accounts.
+ * @param networkInterceptors The interceptors that should be attached to network requests. See
+ * [https://square.github.io/okhttp/interceptors/].
+ * @param interceptors The interceptors that should be attached to all requests. See
+ * [https://square.github.io/okhttp/interceptors/].
+ * @param userAgent The supplemental user agent string which will be added to the internal user agent, identifying the
+ * client.
+ * @param requestTimeoutSeconds The maximum number of seconds a request will wait to complete before timing out.
+ * @param isCertPinningEnabled True if the library should pin certificates, false otherwise. Note: should be turned off
+ * when debugging.
+ * @param logDelegate The instance to which logging will be delegated.
+ * @param logLevel The level of logging which will be performed by the library.
+ * @param cacheDirectory The optional directory in which requests will be cached.
+ * @param cacheMaxSizeBytes The maximum size of the cache, if one was provided.
+ * @param cacheMaxAgeSeconds The maximum age of items in the cache, if one was provided.
+ */
+data class Configuration(
+    val baseUrl: String,
+
+    val clientId: String,
+    val clientSecret: String,
+    val scope: Scopes,
+    val codeGrantRedirectUri: String,
+
+    val locales: List<Locale>,
+
+    val accountStore: AccountStore,
+
+    val networkInterceptors: List<Interceptor>,
+    val interceptors: List<Interceptor>,
+
+    val userAgent: String,
+
+    val requestTimeoutSeconds: Int,
+
+    val isCertPinningEnabled: Boolean,
+
+    val logDelegate: LogDelegate?,
+    val logLevel: LogDelegate.Level,
+
+    val cacheDirectory: File?,
+    val cacheMaxSizeBytes: Long,
+    val cacheMaxAgeSeconds: Int
+) {
+
+    /**
+     * The builder for the [Configuration].
+     *
+     * @param clientId The Vimeo API client ID, required to construct an instance.
+     * @param clientSecret The Vimeo API client secret, required to construct an instance.
+     * @param scope The permission scopes requested by the client, required to construct an instance.
+     */
+    class Builder(private val clientId: String, private val clientSecret: String, private val scope: Scopes) {
+        private var baseUrl: String = DEFAULT_BASE_URL
+
+        private var codeGrantRedirectUrl: String = "vimeo$clientId://auth"
+
+        private var locales: List<Locale> = listOf(Locale.US)
+
+        private var accountStore: AccountStore = InMemoryAccountStore()
+
+        private var networkInterceptors: List<Interceptor> = emptyList()
+        private var interceptors: List<Interceptor> = emptyList()
+
+        private var userAgent: String = DEFAULT_USER_AGENT
+
+        private var requestTimeoutSeconds: Int = DEFAULT_TIMEOUT
+
+        private var isCertPinningEnabled: Boolean = true
+
+        private var logDelegate: LogDelegate? = DefaultLogDelegate()
```

Comment:
Optional: Should we default to logging enabled? I think it would make sense to only enable logging if the consumer chooses to.

Reply:
You're right. We already use other "production-ready" defaults, so it makes sense that logging behaved similarly. However, I think it's be better to set the `LogDelegate.Level` to `NONE` instead of defaulting to no log delegate. If the default `LogDelegate.Level` is `VERBOSE` but the default `LogDelegate` is null, it might be confusing to the consumer why the library isn't logging.